src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2921,6 +2926,29 @@ InfoSub::pointer NetworkOPsImp::addRpcSub (
     return rspEntry;
 }
 
+bool NetworkOPsImp::tryRemoveRpcSub (std::string const& strUrl)
+{
+    ScopedLockType sl (mSubLock);
+    auto pInfo = findRpcSub(strUrl);
+
+    // check to see if any of the stream maps still hold a weak reference to
+    // this entry before removing
+    if ( pInfo &&
+         mSubLedger.find(pInfo->getSeq()) == mSubLedger.end() &&
```

Comment:
This seems somewhat fragile going forward.  Is there a good way to ensure if a new subscription stream type is added to the code that this will be updated? If there isn't a simple way (nothing comes to mind for me), I don't think its worth engineering a solution.